runtime/connectors/blob/blobdownloader.go

```diff
@@ -239,8 +245,24 @@ func (it *blobIterator) plan() ([]*objectWithPlan, error) {
 		return nil, err
 	}
 
-	listOpts := listOptions(it.opts.GlobPattern)
+	listOpts, singleFile := listOptions(it.opts.GlobPattern)
 	it.logger.Info("planner started", zap.String("glob", it.opts.GlobPattern), zap.String("prefix", listOpts.Prefix), observability.ZapCtx(it.ctx))
```

Comment:
Move log message to after the single file case. There will not be a corresponding completed message for single files. For single file, probably have dedicated start/completed messages